src/components/Terminal/ProblemsTab.tsx

```diff
@@ -1,73 +1,288 @@
+      if (severity === DiagnosticSeverity.Information && !governanceShowState.informations) return false;
+      if (severity === DiagnosticSeverity.Hint && !governanceShowState.hints) return false;
+      return true;
+    });
+  }, [diagnostics, governanceShowState.warnings, governanceShowState.informations, governanceShowState.hints]);
 
-  const errorClassName = errors.length ? 'bg-red-500' : 'bg-gray-400';
+  const errorDiagnostics = filteredDiagnostics.filter(diagnostic => diagnostic.severity === 0);
+  const warningDiagnostics = filteredDiagnostics.filter(diagnostic => diagnostic.severity === 1);
+  const infoDiagnostics = filteredDiagnostics.filter(diagnostic => diagnostic.severity === 2);
+  const hintDiagnostics = filteredDiagnostics.filter(diagnostic => diagnostic.severity === 3);
 
   return (
-    <div>
-      <span className="text-xs">Problems</span>
-      <span className={`inline-block rounded-full px-1.5 py-0.5 ml-1.5 -mt-2 text-xs ${errorClassName}`}>
-        {errors.length || 0}
-      </span>
+    <div className='flex flex-row items-center'>
+      <span className="text-xs">Diagnostics</span>
+      <ul className='flex flex-row items-center'>
+        {errorDiagnostics.length > 0 && (
+          <li>
+            <span className='inline-block rounded-full px-1.5 ml-1.5 text-xs bg-red-500'>
+              {errorDiagnostics.length}
+            </span>
+          </li>
+        )}
+        {warningDiagnostics.length > 0 && (
+          <li>
+            <span className='inline-block rounded-full px-1.5 ml-1 text-xs bg-yellow-500'>
+              {warningDiagnostics.length}
+            </span>
+          </li>
+        )}
+        {infoDiagnostics.length > 0 && (
+          <li>
+            <span className='inline-block rounded-full px-1.5 ml-1 text-xs bg-blue-500'>
+              {infoDiagnostics.length}
+            </span>
+          </li>
+        )}
+        {hintDiagnostics.length > 0 && (
+          <li>
+            <span className='inline-block rounded-full px-1.5 ml-1 text-xs bg-blue-500'>
+              {hintDiagnostics.length}
+            </span>
+          </li>
+        )}
+      </ul>
+    </div>
+  );
+};
+
+interface SeverityIconProps {
+  severity: Diagnostic['severity']
+}
+
+const SeverityIcon: React.FunctionComponent<SeverityIconProps> = ({ severity }) => {
+  switch (severity) {
+  case 1: return (
+    <div className='flex flex-row items-center justify-center'>
+      <VscWarning className='text-yellow-500 w-4 h-4' />
+    </div>
+  );
+  case 2: return (
+    <div className='flex flex-row items-center justify-center'>
+      <VscInfo className='text-blue-500 w-4 h-4' />
+    </div>
+  );
+  case 3: return (
+    <div className='flex flex-row items-center justify-center'>
+      <VscLightbulb className='text-green-500 w-4 h-4' />
+    </div>
+  );
+  default: return (
+    <div className='flex flex-row items-center justify-center'>
+      <VscError className='text-red-500 w-4 h-4' />
     </div>
```

Comment:
It would be awesome to have tooltips in these icons. Until I came to the code, I couldn't figure out what the green light bulb meant.

Reply:
I hope that you like it :)

![image](https://user-images.githubusercontent.com/20404945/200800838-0eec9aed-205a-4f8d-b7cd-e431a7df4aad.png)